core/io/resource.cpp

```diff
@@ -564,6 +575,8 @@ void Resource::_bind_methods() {
 
 	::ClassDB::add_virtual_method(get_class_static(), get_rid_bind, true, Vector<String>(), true);
 	GDVIRTUAL_BIND(_setup_local_to_scene);
+	GDVIRTUAL_BIND(_reset_state);
+	GDVIRTUAL_BIND(_set_path_cache, "path");
```

Comment:
I'm not sure this should be virtual.

Reply:
Hi @RandomShaper, I made these virtual so that classes created in GDExtension that are derived from `Resource` were able to perform some actions on `reset_state` and `set_path_cache` without disrupting the behavior in the super type in the engine.  If there is a better way, any pointer?

Reply:
For `reset_state`, I see the clear point. My concern was about `set_path_cache`. I'm not sure any resource type should ever do anything about it because it's serves an internal purpose. In other words, it's mostly a quirk. That said, well, it's already virtual in C++ so it's only fair that it's virtual GDExtension-wise as well.

Reply:
Right, it's mostly the philosophical stance of trying to align GDE with C++ modules.

Reply:
@RandomShaper do you think we can try and target 4.4 with this?

Reply:
Sorry, I missed this question, but, well, you can see it's already on its way to 4.4. 😀